ports/rp2/machine_spi.c

```diff
@@ -167,7 +179,10 @@ mp_obj_t machine_spi_make_new(const mp_obj_type_t *type, size_t n_args, size_t n
         }
         self->mosi = mosi;
     }
-    if (args[ARG_miso].u_obj != mp_const_none) {
+
+    if (args[ARG_miso].u_obj == mp_const_none) {
+        self->miso = MICROPY_HW_SPI_PIN_UNUSED;
+    } else {
```

Comment:
I think this line needs to be `} else if (args[ARG_miso].u_obj != MP_OBJ_NEW_SMALL_INT(-1)) {`.

Otherwise in the default case (miso not specified) it'll try to access pin `-1` and raise an exception.

Reply:
Ooof, you are correct- I think this got lost in the chaos of the other PR.

Indeed -1 is the new "not specified" and `None` is "do not use."

Fixed.